### DISCUSSION TITLE
3.0.0-beta1

### DISCUSSION THREAD

**Discussion by akallabeth** [FreeRDP/FreeRDP#9191] [Q&A]
[freerdp-3.0.0-beta1.zip](https://github.com/FreeRDP/FreeRDP/files/12129474/freerdp-3.0.0-beta1.zip)
[freerdp-3.0.0-beta1.tar.gz](https://github.com/FreeRDP/FreeRDP/files/12129476/freerdp-3.0.0-beta1.tar.gz)
We are pleased to announce the first beta release for the next stable 3.0
 series of FreeRDP. It has been a huge endeavour to implement all the new
 shiny bells and whistles as well as clean up the code base and we´re still
 ironing out some smaller glitches.
 This is the first API breaking change since the 2.0 series and there are
 some adjustments to be made for existing applications.
 See https://github.com/FreeRDP/FreeRDP/wiki/FreeRDP3-migration-notes for
 help (still incomplete)
 
 Noteworthy changes:
 * Support for AAD/AVD authentication
 * Support for websocket transport
 * Support smartcard authentication (TLS and NLA)
 * Full smartcard emulation support (login with certificate + key)
 * Rewritten proxy, new module API
 * New reference client based on SDL2 (work in progress)
 * Rewritten logging, now parsing issues are all writing to the log so
   that issues with protocol incompatibilities can be easier analyzed
   by just turning on logging
 * Full OpenSSL 3 support
 * Internal implementations for RC4, MD4 and MD5 (required for non critical
   parts in RDP but not part of more recend SSL libraries)
 * Updated RDP protocol support
 * Improved xfreerdp remote app support
 * Reworked internal state machine for both client and server implementations
 * Server implementations can now make use of connect-time network autodetection
 * Improved clipboard handling, now also support server-to-client file transfer
   (currently xfreerdp only)
 * EnhancedRemoteApp support: Utilizing the more modern standard allows remote
   apps with less glitches and window shadows
 * Added client- and server-side handling for RDSTLS
 * Support for the graphics redirection channel
 
 For a complete and detailed change log since the last release run:
 git log 3.0.0-beta1..2.10.0

<hr /><em>This discussion was created from the release <a href='https://github.com/FreeRDP/FreeRDP/releases/tag/3.0.0-beta1'>3.0.0-beta1</a>.</em>

**Comment by designchris** [comment]
What does "Support for the graphics redirection channel" address? Is this about support for [MS-RDPECAM]?

**Reply by akallabeth** [reply]
it is the channel in `channels/gfxredir` microsoft is using for WSL2 (it does not have an official spec as far as I know, but that might change at some point)

**Reply by designchris** [reply]
Thanks for pointing that out!

**Comment by zshuaia** [comment]
"EnhancedRemoteApp support: Utilizing the more modern standard allows remote
apps with less glitches and window shadows"

Is the scope of the change extensive? Is there any special topic or documentation on this?
Is there a guideline for incorporating this feature from version 2.x？

**Reply by akallabeth** [reply]
* it utilizes [[RDPEGFX]](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpegfx/da5c75f9-cd99-450c-98c4-014a496942b0) for graphics transport
* [[MS-RDPERP]](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdperp/ead02bce-64ef-41d6-aa06-8565956b9fe7) documents the protocol details
* no, as we do not plan to backport it.
